api/envoy/extensions/transport_sockets/tls/v3/tls.proto

```diff
@@ -158,4 +180,8 @@ message CommonTlsContext {
   //
   // There is no default for this parameter. If empty, Envoy will not expose ALPN.
   repeated string alpn_protocols = 4;
+
+  // Config for fetching TLS certificates via CA client plugin.
+  // [#not-implemented-hide:]
+  CAClientPluginConfig tls_certificate_ca_client_plugin = 9;
```

Comment:
Does it make sense to make the different fields for getting the `tls_certificates` also a `oneof`? If so, could we mark them with appropriate annotation to promote to oneof as well.

Reply:
Yes, I think I will go ahead and just do that for all oneof candidates.

Reply:
Thinking more about it, I see that they already had `oneof validation_context_type` to select only one among the 3 existing options (before I added the 4th one). But they didn't do the same for  `tls_certificates` and `tls_certificate_sds_secret_configs` which could have been put in `oneof`.

@htuch could you throw some light on this? Should my `field_migrate.oneof_promotion` annotation include all 3 (`tls_certificates`,  `tls_certificate_sds_secret_configs` and `tls_certificate_ca_client_plugin`) or just the last 2 or you think `oneof` is not warranted here? And the reason for that?